src/client/components/Meal.js

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import { useParams } from 'react-router-dom';
+import {useState} from "react";
+import Alert from 'react-popup-alert'
+
+
+const API_URL="/api";
+
+
+function Meal({oneMeal , updateReservation, allReservations}){
+
+  let params = useParams();
+  let meal = oneMeal(parseInt(params.id));
+
+  const [alert, setAlert] =useState({
+    type: 'error',
+    text: 'This is a alert message',
+    show: false
+  })
+
+  function onCloseAlert() {
+    setAlert({
+      type: '',
+      text: '',
+      show: false
+    })
+  }
+
+  function onShowAlert(type) {
+    setAlert({
+      type: type,
+      text: 'Demo alert',
+      show: true
+    })
+  }
+
+
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    const url = `${API_URL}/reservations/`;
+    return fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({number_of_guests:parseInt(numberOfGuests),meal_id:parseInt(params.id),contact_phonenumber:phoneNumber,contact_name:name,contact_email:email}),
+    })
+      .then((reponse) => reponse.json())
+      .then(() => updateReservation(response));
+  }
+  function getReservations(mealid) {
+
+    let numberOfPeople = 0;
+   for (let i = 0; i < allReservations.length; i++) {
+   if(allReservations[i].id == mealid) {
+  numberOfPeople += allReservations[i].numberOfPeople
+}
+   }
+   return numberOfPeople < meal.max_reservations;
+ }
+ 
+  const [numberOfGuests ,setnumberOfGuests]=useState("");
+  const [phoneNumber,setPhoneNumber]=useState("");
+  const [name,setName]=useState("");
+  const [email,setEmail]=useState("");
+
+  const handlePhoneNumberChange = (event) => setPhoneNumber(event.target.value);
+  const handleNameChange = (event) => setName(event.target.value);
+  const handleEmailChange = (event) => setEmail(event.target.value);
+const handleNumberOfGuests=(event) => setnumberOfGuests(event.target.value);
+  
+return(
+  <>
+  
+  <button onClick={() => onShowAlert('success')}>
+    testbbbbutton
+  </button>
+    <div>
+        <p>{meal.title}</p>
+        <p>{meal.description}</p>
+        
+    </div>
+    {getReservations(meal.id) &&
+    <div>
+      <h1>Submit a new meal!</h1>
```

Comment:
Hm maybe 'Submit a new meal!' is a bit misleading? Cause you're adding a new reservation for a meal not a new meal right?